ash/src/vk/prelude.rs

```diff
@@ -56,3 +56,11 @@ impl From<vk::Extent2D> for vk::Rect2D {
         }
     }
 }
+
+/// Structures implementing this trait are layout-compatible with [`vk::BaseInStructure`] and
+/// [`vk::BaseOutStructure`]. That is, it has a `p_next` pointer field to chain other
+/// [`TaggedStructure`]s and an `s_type` field indicating its type, which must always match the
+/// value of [`TaggedStructure::STRUCTURE_TYPE`].
```

Comment:
This isn't a complete definition of layout-compatibility, and might confuse an inexperienced reader. If we want to provide a definition, we should also more clearly note the field ordering, and that the struct must be `repr(C)`.

Reply:
The main point here was stating that it must have the same layout as `Base*Structure`, then proceeding to explain that `s_type` is what's used here to be equal to `TaggedStructure`. Perhaps it is better to remove ``That is, it has a `p_next` pointer field to chain other [`TaggedStructure`]s`` and only mention `s_type` explicitly?

Reply:
I understand the use of "that is, it has" here to specifically present a definition. Replacing it with "Such structures have" seems to more clearly describe properties of them without necessarily defining them in full, at which point I'm indifferent to including any discussion of p_next.